File: tests/fixtures/multi-language/sample.java
// Java 测试文件
package com.example.user;

import java.util.HashMap;
import java.util.Map;
import java.util.Optional;
import java.util.regex.Pattern;

public class UserService {
    private final Map<String, User> users = new HashMap<>();
    private static final Pattern EMAIL_PATTERN = 
        Pattern.compile("^[^\\s@]+@[^\\s@]+\\.[^\\s@]+$");

    public Optional<User> getUser(String id) {
        return Optional.ofNullable(users.get(id));
    }

    public void createUser(User user) {
        users.put(user.getId(), user);
    }

    public boolean deleteUser(String id) {
        return users.remove(id) != null;
    }

    public static boolean validateEmail(String email) {
        return EMAIL_PATTERN.matcher(email).matches();
    }
}

class User {
    private final String id;
    private final String name;
    private final String email;

    public User(String id, String name, String email) {
        this.id = id;
        this.name = name;
        this.email = email;
    }

    public String getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getEmail() {
        return email;
    }
}

File: tests/fixtures/multi-language/sample.py
# Python 测试文件
from typing import Optional, Dict
from dataclasses import dataclass

@dataclass
class User:
    id: str
    name: str
    email: str

class UserService:
    def __init__(self):
        self.users: Dict[str, User] = {}
    
    async def get_user(self, user_id: str) -> Optional[User]:
        """获取用户信息"""
        return self.users.get(user_id)
    
    async def create_user(self, user: User) -> None:
        """创建新用户"""
        self.users[user.id] = user
    
    async def delete_user(self, user_id: str) -> bool:
        """删除用户"""
        if user_id in self.users:
            del self.users[user_id]
            return True
        return False

def validate_email(email: str) -> bool:
    """验证邮箱格式"""
    import re
    pattern = r'^[^\s@]+@[^\s@]+\.[^\s@]+$'
    return bool(re.match(pattern, email))
